packages/extension/src/newtab/ShortcutLinks.tsx

```diff
@@ -101,31 +215,74 @@ export default function ShortcutLinks({
     });
   };
 
+  const onLinkClick = () => {
+    trackEvent({
+      event_name: AnalyticsEvent.Click,
+      target_type: TargetType.Shortcuts,
+      extra: JSON.stringify({ source: shortcutSource }),
+    });
+  };
+
   return (
     <>
-      {shortcutLinks?.length ? (
-        <CustomLinks
-          links={shortcutLinks}
-          className={className}
-          onOptions={onOptionsOpen}
-          onLinkClick={() => {
-            trackEvent({
-              event_name: AnalyticsEvent.Click,
-              target_type: TargetType.Shortcuts,
-              extra: JSON.stringify({ source: shortcutSource }),
-            });
-          }}
-        />
-      ) : (
-        <Button
-          className={className}
-          variant={ButtonVariant.Tertiary}
-          icon={<PlusIcon />}
-          iconPosition={ButtonIconPosition.Right}
-          onClick={onOptionsOpen}
+      {isShortcutsV1 ? (
+        <div
+          className={classNames(
+            'hidden tablet:flex',
+            shouldUseMobileFeedLayout ? 'mx-6 mb-3 mt-1' : '-mt-2 mb-5',
+          )}
         >
-          Add shortcuts
-        </Button>
+          {shortcutLinks?.length ? (
+            <>
+              {shortcutLinks.map((url) => (
+                <ShortcutV1Item key={url} url={url} onLinkClick={onLinkClick} />
+              ))}
+              <Button
+                variant={ButtonVariant.Tertiary}
+                size={ButtonSize.Small}
+                icon={<MenuIcon className="rotate-90" secondary />}
+                onClick={onMenuClick}
+                className="mt-2"
+              />
+            </>
+          ) : (
+            <>
+              <ShortCutV1Placeholder initialItem onClick={onOptionsOpen} />
+              {Array.from({ length: 5 }).map((_, index) => (
+                /* eslint-disable-next-line react/no-array-index-key */
+                <ShortCutV1Placeholder key={index} onClick={onOptionsOpen} />
+              ))}
+              <Button
+                variant={ButtonVariant.Tertiary}
+                onClick={toggleShowTopSites}
+                size={ButtonSize.Small}
+                icon={<ClearIcon secondary />}
+                className="mt-2"
+              />
+            </>
+          )}
+        </div>
+      ) : (
```

Comment:
Can we move these inner components + conditions in an other named component that maybe it's fine to be in this file but makes it easier to read?

Valid for conditions below as well

Reply:
I had that, but it failed the tests because of too many re-renders 😅

Reply:
Maybe `withExperiment` can help to just split out the components into two and we don't need to test the experimented ones?

Reply:
@capJavert // @nensidosari Jest really doesn't like the existing one as inline. component.
I did the v1 as one, hopefully that's ok for now?

Also removed the default value in favor of abstracting logic to component itself.